platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/TaskScheduler.java

```diff
@@ -0,0 +1,174 @@
+ */
+public abstract class TaskScheduler<OUT> extends OutputWire<OUT> {
+
+    private final boolean flushEnabled;
+
+    /**
+     * Constructor.
+     *
+     * @param model               the wiring model containing this task scheduler
+     * @param name                the name of the task scheduler
+     * @param flushEnabled        if true, then {@link #flush()} will be enabled, otherwise it will throw.
+     * @param insertionIsBlocking when data is inserted into this task scheduler, will it block until capacity is
+     *                            available?
+     */
+    protected TaskScheduler(
+            @NonNull final WiringModel model,
+            @NonNull final String name,
+            final boolean flushEnabled,
+            final boolean insertionIsBlocking) {
+        super(model, name, true, insertionIsBlocking);
+        this.flushEnabled = flushEnabled;
+    }
+
+    /**
+     * Cast this scheduler into whatever a variable is expecting. Sometimes the compiler gets confused with generics,
+     * and path of least resistance is to just cast to the proper data type.
+     *
+     * <p>
+     * Warning: this will appease the compiler, but it is possible to cast a scheduler into a data type that will cause
+     * runtime exceptions. Use with appropriate caution.
+     *
+     * @param <X> the type to cast to
+     * @return this, cast into whatever type is requested
+     */
+    @NonNull
+    @SuppressWarnings("unchecked")
+    public final <X> TaskScheduler<X> cast() {
+        return (TaskScheduler<X>) this;
+    }
+
+    /**
+     * Get an input wire for passing data to this task scheduler. In order to use this wire, a handler must be bound via
+     * {@link InputWire#bind(Consumer)}.
+     *
+     * @param name the name of the input wire
+     * @param <I>  the type of data that is inserted via this input wire
+     * @return the input wire
+     */
+    @NonNull
+    public final <I> InputWire<I, OUT> buildInputWire(@NonNull final String name) {
+        return new InputWire<>(this, name);
+    }
+
+    /**
+     * Get the number of unprocessed tasks. Returns -1 if this task scheduler is not monitoring the number of
+     * unprocessed tasks. Schedulers do not track the number of unprocessed tasks by default. This method will always
+     * return -1 unless one of the following is true:
+     * <ul>
+     * <li>{@link TaskSchedulerMetricsBuilder#withUnhandledTaskMetricEnabled(boolean)} is called with the value true</li>
+     * <li>{@link TaskSchedulerBuilder#withUnhandledTaskCapacity(long)} is passed a positive value</li>
+     * <li>{@link TaskSchedulerBuilder#withOnRamp(ObjectCounter)} is passed a counter that is not a no op counter</li>
+     * </ul>
+     */
+    public abstract long getUnprocessedTaskCount();
+
+    /**
+     * Flush all data in the task scheduler. Blocks until all data currently in flight has been processed.
```

Comment:
Is this correct?
```suggestion
     * Flush all data in the task scheduler. Blocks until all data currently in flight has been scheduled.
```

Reply:
"processed" is actually correct here I think. We will actually wait for all processing to be fully complete before this method returns. Waiting for data to be scheduled implies to me that it may be processed at a future time.